src/main/java/com/fauna/codec/FaunaType.java

```diff
@@ -0,0 +1,16 @@
+package com.fauna.codec;
+
+public enum FaunaType {
+    Int,
+    Long,
+    Double,
+    String,
+    Date,
+    Time,
+    Boolean,
+    Object,
+    Ref,
+    Document,
+    Array,
+    Bytes, Null, Stream, Module, Set, Unknown
```

Comment:
Suggestion: `Unknown` is technically not a type. Removing it from here and making `getFaunaType()` return an `Optional` (or fail with an exception) would make this enum more precise.